### PR TITLE
En 5237 check if nil reflect

### PR DESCRIPTION
**Pull request by traian-schiau**
IfNilReflect

### CODE REVIEW

File: core/check/ifNil.go [hunk 1]
```diff
@@ -1,9 +1,25 @@
...
+
 // IfNil tests if the provided interface pointer or underlying object is nil
 func IfNil(checker NilInterfaceChecker) bool {
 	if checker == nil {
 		return true
 	}
 	return checker.IsInterfaceNil()
 }
+
+// IfNil tests if the provided interface pointer or underlying pointer receiver is nil
```
Comment (miiu96) [inline core/check/ifNil.go]: IfNilReflect tests...

File: core/check/ifNil.go [hunk 2]
```diff
@@ -1,9 +1,25 @@
...
+func IfNilReflect(i interface{}) bool {
+	if v := reflect.ValueOf(i); v.IsValid() {
+		if v.Kind() == reflect.Ptr && v.IsNil() {
+			return true
+		}
+		return false
+	} else {
```
Comment (miiu96) [inline core/check/ifNil.go]: maybe you can remove **else**

Comment (sasurobert) [reply core/check/ifNil.go]: no need for else. return true

File: core/check/ifNil.go (hunk 1 above)
Comment (sasurobert) [reply core/check/ifNil.go]: +1

Comment (traian-schiau) [reply core/check/ifNil.go]: done

File: core/check/ifNil.go (hunk 2 above)
Comment (traian-schiau) [reply core/check/ifNil.go]: done

**Pull request merged**